src/librustdoc/html/render/search_index.rs

```diff
@@ -201,7 +201,7 @@ pub(crate) fn build_index(
                             // exported from this same module). It's also likely to Do
                             // What I Mean, since if a re-export changes the name, it might
                             // also be a change in semantic meaning.
-                            .filter(|fqp| fqp.last() == fqp.last());
+                            .filter(|this_fqp| this_fqp.last() == fqp.last());
```

Comment:
there's another filter that's the inverse of this at line 212

Reply:
Not actually the same. This one only checks the last path component. The one on 212 checks the entire path.

Reply:
ahhh indeed